commands/template_store_list.go

```diff
@@ -0,0 +1,170 @@
+package commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// The URL where the store resides
+	DefaultTemplateList = "https://raw.githubusercontent.com/martindekov/store/martindekov/unofficial_templates_json/templates.json"
+	incubatorRepo       = "openfaas-incubator"
+	officialRepo        = "openfaas"
+	allTemplates        = "all"
+)
+
+var (
+	source string
+)
+
+func init() {
+	templateStoreListCmd.Flags().StringVarP(&source, "source", "s", allTemplates, "The source of the templates")
+	templateStoreListCmd.Flags().BoolVarP(&verbose, "verbose", "v", false, "Shows additional language and platform")
+
+	templateStoreCmd.AddCommand(templateStoreListCmd)
+}
+
+var templateStoreListCmd = &cobra.Command{
+	Use:     `list`,
+	Short:   `List templates from OpenFaaS organizations`,
+	Aliases: []string{"ls"},
+	Long:    `List templates supported by openfaas or openfaas-incubator organizations or both`,
+	Example: `  faas-cli template store list --source=openfaas --verbose
+  faas-cli template store ls --source=openfaas-incubator -v
+  faas-cli template store ls --source=all`,
+	RunE: runTemplateStoreList,
+}
+
+func runTemplateStoreList(cmd *cobra.Command, args []string) error {
+	templatesInfo, templatesErr := getTemplateInfo(DefaultTemplateList)
+	if templatesErr != nil {
+		return fmt.Errorf("error while getting templates info: %s\n", templatesErr)
+	}
+
+	formattedOutput := formatTemplatesOutput(templatesInfo)
+
+	fmt.Println(formattedOutput)
```

Comment:
I would fprint this to `cmd.OutOrStdout()` this returns a writer, typically stdout, but it makes it easier to test the cmd output.

Reply:
Yeah seems perfect, haven't though of it that way

Reply:
This is now done ^